src/screens/AddWalletSheet.tsx

```diff
@@ -167,107 +206,208 @@ export const AddWalletSheet = () => {
       });
     } catch (e) {
       setIsWalletLoading(null);
-      logger.log('Error while trying to add account', e);
+      logger.error(e as RainbowError, {
+        description: 'Error while trying to add account',
+      });
     }
   };
 
-  const onPressRestore = () => {
+  const onPressRestoreFromSeed = () => {
     analytics.track('Tapped "Add an existing wallet"');
     analyticsV2.track(analyticsV2.event.addWalletFlowStarted, {
-      isFirstWallet: false,
+      isFirstWallet,
       type: 'seed',
     });
     navigate(Routes.ADD_WALLET_NAVIGATOR, {
-      screen: Routes.IMPORT_SEED_PHRASE_FLOW,
+      screen: Routes.IMPORT_OR_WATCH_WALLET_SHEET,
       params: { type: 'import' },
     });
   };
 
   const onPressWatch = () => {
     analytics.track('Tapped "Add an existing wallet"');
     analyticsV2.track(analyticsV2.event.addWalletFlowStarted, {
-      isFirstWallet: false,
+      isFirstWallet,
       type: 'watch',
     });
     navigate(Routes.ADD_WALLET_NAVIGATOR, {
-      screen: Routes.IMPORT_SEED_PHRASE_FLOW,
+      screen: Routes.IMPORT_OR_WATCH_WALLET_SHEET,
       params: { type: 'watch' },
     });
   };
 
+  const onPressRestoreFromCloud = async () => {
+    analyticsV2.track(analyticsV2.event.addWalletFlowStarted, {
+      isFirstWallet,
+      type: 'seed',
+    });
+    if (IS_ANDROID) {
+      const isAvailable = await isCloudBackupAvailable();
+      if (isAvailable) {
+        let proceed = false;
+        try {
+          const data = await fetchUserDataFromCloud();
+          if (data?.wallets) {
+            Object.values(data.wallets as RainbowWallet[]).forEach(wallet => {
+              if (
+                wallet.backedUp &&
+                wallet.backupType === WalletBackupTypes.cloud
+              ) {
+                proceed = true;
+              }
+            });
+            if (proceed) {
+              navigate(Routes.RESTORE_SHEET, { userData: data });
+            }
+          }
+          logger.info(`Downloaded ${cloudPlatform} backup info`);
+        } catch (e) {
+          logger.info((e as Error).message);
```

Comment:
Since you updated to use the new logger (thank youuuu) this should be `logger.error`